### PR TITLE
Ensure ilm expiry creates non null deletemarker for versioned bucket

### PR DESCRIPTION
**Pull request by poornas**
Fixes regression from #15521

## How to test this PR?
create `mc ilm add alias/bucket --expiry-days=1` where bucket is versioned.
upload objects to this bucket
after expiry date reached `mc stat alias/bucket/obj` ends up creating null delete marker with master and deletemarker version with this fix

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] Fixes a regression (If yes, please add `commit-id` or `PR #` here) #15521
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fminio%2fpull%2fNNNNN)

### CODE REVIEW

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Review by harshavardhana — CHANGES_REQUESTED** [review]
The PR looks good, can you add a functional test for this? - I think there is one for idempotent Delete() that can be extended.

File: cmd/erasure-object.go [hunk 1]
```diff
@@ -1573,13 +1575,16 @@ func (er erasureObjects) DeleteObject(ctx context.Context, bucket, object string
 					Object: object,
 				}
 			}
+			markDelete = goi.VersionID != ""
 		}
 	}
 
 	versionFound := !(opts.DeleteMarker && opts.VersionID != "")
 
 	// Determine whether to mark object deleted for replication
```
Comment (vadmeste) [inline cmd/erasure-object.go]:
```suggestion

```

File: cmd/erasure-object.go [hunk 2]
```diff
@@ -1549,6 +1549,8 @@ func (er erasureObjects) DeleteObject(ctx context.Context, bucket, object string
 
 	storageDisks := er.getDisks()
 
+	var markDelete bool
```
Comment (vadmeste) [inline cmd/erasure-object.go:1553]:
```suggestion
	// Determine whether to mark object deleted for replication
	var markDelete bool
```

**Review by vadmeste — COMMENTED** [review]
One comment

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Review by vadmeste — APPROVED** [review]
LGTM

**Comment by poornas** [comment]
> The PR looks good, can you add a functional test for this? - I think there is one for idempotent Delete() that can be extended.

don't see how we can do a functional test here since ilm expiry has a wait of 24 hrs.

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**